src/adapters/ip_camera/api.rs

```diff
@@ -62,12 +62,18 @@ pub struct IpCamera {
 }
 
 impl IpCamera {
-    pub fn new(udn: &str, url: &str, root_snapshot_dir: &str) -> Self {
-        IpCamera {
+    pub fn new(udn: &str, url: &str, root_snapshot_dir: &str) -> Result<Self, Error> {
+        let camera = IpCamera {
             udn: udn.to_owned(),
             url: url.to_owned(),
             snapshot_dir: format!("{}/{}", root_snapshot_dir, udn)
+        };
+        // Create a directory to store snapshots for this camera.
+        if let Err(err) = fs::create_dir_all(&camera.snapshot_dir) {
```

Comment:
create_dir_all() also fails if the directory already exists. That should be just ignored, no? We do that in https://github.com/fxbox/foxbox/blob/master/src/profile_service.rs#L61 for instance.